docs/scripting-element-object.rst

```diff
@@ -0,0 +1,566 @@
+
+**Returns:** ``ok, value`` pair. If ``ok`` is nil then error happened during the function call;
+``value`` provides an information about error type; otherwise ``value`` is a value of the
+element.
+
+**Async:** no.
+
+The value can be a
+* string - for text/radio input and select and other element types
+* array of strings - for multi select
+* boolean - for checkbox input
+
+
+.. _splash-element-form-values:
+
+element:form_values
+-------------------
+
+Return a table with form values if the element type is *form*
+
+**Signature:** ``ok, info = element:form_values()``
+
+**Returns:** ``ok, values`` pair. If ``ok`` is nil then error happened during the function call
+or node type is not *form*; ``values`` provides an information about error type; otherwise
+``values`` is a table of values.
+
+**Async:** no.
+
+
+.. _splash-element-fill:
+
+element:fill
+------------
+
+Fill the form with the provided values
+
+**Signature:** ``ok, reason = element:fill(values)``
+
+**Parameters:**
+
+* values - table with input names as keys and values as input values
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+In order to fill your form your inputs must have ``name`` property and this method will
+select those input using that property.
+
+Example of filling the following form:
+
+.. code-block:: html
+
+    <form id="login">
+        <input type="text" name="username" />
+        <input type="password" name="password" />
+    </form>
+
+.. code-block:: lua
+
+    function main(splash)
+        assert(splash:select('.login'):fill({ username="admin", password="pass" }))
+    end
+
+
+.. _splash-element-send-keys:
+
+element:send_keys
+-----------------
+
+Send keyboard events to the element.
+
+**Signature:** ``ok = element:send_keys(keys)``
+
+**Parameters**
+
+* keys - string representing the keys to be sent as keyboard events.
+
+**Returns:** ``ok`` pair. If ``ok`` is nil then error happened during the function call.
+
+**Async:** no.
+
+This methods do the following:
+
+* clicks on the element
+* send the specified keyboard events
+
+See more about keyboard events in in :ref:`splash-send-keys`.
+
+
+.. _splash-element-send-text:
+
+element:send_text
+-----------------
+
+Send keyboard events to the element.
+
+**Signature:** ``ok = element:send_text(text)``
+
+**Parameters**
+
+* text - string to be sent as input.
+
+**Returns:** ``ok`` pair. If ``ok`` is nil then error happened during the function call.
+
+**Async:** no.
+
+This methods do the following:
+
+* clicks on the element
+* send the specified text to the element
+
+See more about it in :ref:`splash-send-text`.
+
+
+
+.. _splash-element-submit:
+
+element:submit
+--------------
+
+Submit the form element.
+
+**Signature:** ``ok = element:submit()``
+
+**Returns:** ``ok`` pair. If ``ok`` is nil then error happened during the function call.
```

Comment:
The reason is absent my mistake. The reason can be if you want to submit non-form element.